src/v/kafka/server/handlers/fetch.cc

```diff
@@ -418,7 +418,7 @@ read_result::memory_units_t reserve_memory_units(
 static void fill_fetch_responses(
   op_context& octx,
   std::vector<read_result> results,
-  std::vector<op_context::response_placeholder_ptr> responses,
+  std::vector<op_context::response_placeholder_ptr>& responses,
```

Comment:
const ref ?

Reply:
Nice catch, will switch it to const.

Reply:
@ballard26 can you comment a bit more on this change? Seems like a good perf win, should it apply also to results?

Reply:
`responses` can definitely be a const ref. Nice catch. 

As for `results` I don't think we'd want to. `responses` is never modified and is always the same for every call to `fill_fetch_responses`. Therefore can be passed by reference and easily re-used. But with `results` `fill_fetch_responses` moves data out of it. And even if it didn't `fetch_ntps_in_parallel` always returns a new `results` vector so there is no re-use that can occur anyways.